File: LeetCode_3304/Main.cpp
#include<bits/stdc++.h>
using namespace std;


class Solution {
public:
    char kthCharacter(int k) {
        string cur = "a";
        while (cur.size() < k) {
            string temp = cur;
            for (int i = 0; i < cur.size(); i++) {
                int d = cur[i] - 'a';
                d = (d + 1) % 26;
                temp[i] = d + 'a';
            }
            cur += temp;
        }
        return cur[k - 1];
    }
};

int main() {

    return 0;
}

File: LeetCode_3304/Main.java
class Solution {
    public char kthCharacter(int k) {
        StringBuilder cur = new StringBuilder("a");

        while (cur.length() < k) {
            StringBuilder temp = new StringBuilder();

            for (int i = 0; i < cur.length(); i++) {
                int d = cur.charAt(i) - 'a';
                d = (d + 1) % 26;
                temp.append((char) (d + 'a'));
            }

            cur.append(temp);
        }

        return cur.charAt(k - 1);
    }
}

public class Main {
    public static void main(String[] args) {
        
    }
}
